src/components/ResetScaleStep.js

```diff
@@ -9,26 +9,23 @@ type Props = {|
 
 type State = {|
   resetWeight: string,
-  inputError: boolean,
 |};
 
 class ResetScaleStep extends PureComponent<Props, State> {
   state = {
     resetWeight: '',
-    inputError: false,
   };
 
   completeStep = (event: SyntheticEvent<HTMLFormElement>) => {
     event.preventDefault();
-    const { onCompleted } = this.props;
     const { resetWeight } = this.state;
-    if (resetWeight) {
-      onCompleted(Number(resetWeight));
-    } else {
-      this.setState({
-        inputError: true,
-      });
+    if (!resetWeight) {
+      this.setState({ resetWeight: '0' }, () => this.completeStep(event));
```

Comment:
Not sure if we need to call `this.completeStep` a second time and do the empty return? Because of the default weight on line 28 (`onCompleted(Number(resetWeight || 0))`)  it should suffice to just set the state and continue (even though setState might be batched/happen async)